tensorly/testing.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+
+from tensorly import backend as T
+
+
+def assert_array_equal(a, b, *args, **kwargs):
+    np.testing.assert_array_equal(T.to_numpy(a), T.to_numpy(b),
+                                  *args, **kwargs)
+
+
+def assert_array_almost_equal(a, b, *args, **kwargs):
+    np.testing.assert_array_almost_equal(T.to_numpy(a), T.to_numpy(b),
+                                         *args, **kwargs)
+
+
+def assert_equal(actual, desired, *args, **kwargs):
+    def _tensor_to_numpy(x):
```

Comment:
Minor point: maybe this test (checking whether the tensor is one before converting to numpy) should be in the backend.to_numpy?

Reply:
Perhaps, but that becomes tricky because what should `tl.to_numpy([1, 2, 3])` do? It's not a tensor, but the user probably wants it to be an array (and currently `tl.to_numpy` does this, whereas the implementation below won't). I opted to not change the implementation of any of the backend functions in this large PR - future changes could be discussed later.

Reply:
Agreed.